query-service-impl/src/main/java/org/hypertrace/core/query/service/ExecutionContext.java

```diff
@@ -35,14 +38,83 @@
   // while the selectedColumns
   // is a set of column names.
   private final LinkedHashSet<Expression> allSelections;
+  private final Map<String,String> timeSeriesColumnMap;
+  private final Map<String,List<Long>> timeFilterMap;
 
   public ExecutionContext(String tenantId, QueryRequest request) {
     this.tenantId = tenantId;
     this.selectedColumns = new LinkedHashSet<>();
     this.allSelections = new LinkedHashSet<>();
+    this.timeSeriesColumnMap = new HashMap<>();
+    this.timeFilterMap = new HashMap<String,List<Long>>();
+    analyzeForAvgRate(request);
     analyze(request);
   }
 
+  private void analyzeForAvgRate(QueryRequest request){
+    setTimeSeriesColumnMap(request);
+    setTimeFilterMap(request);
+  }
+
+  private void setTimeSeriesColumnMap(QueryRequest request){
+
+    if(request.getGroupByCount() > 0){
+      for(Expression expression : request.getGroupByList()){
+        if(expression.getValueCase() == ValueCase.FUNCTION && expression.getFunction().getFunctionName().equals("dateTimeConvert")) {
+
+          String column = null;
+          String period = null;
+
+          //assuming only one column and one literal (with period in seconds) is present in one dateTimeConvert groupBy
+          for(Expression childExpression : expression.getFunction().getArgumentsList()) {
+            if(childExpression.getValueCase() == ValueCase.COLUMNIDENTIFIER) {
+              column = childExpression.getColumnIdentifier().getColumnName().split("[.]")[0];
+            }
+            if(childExpression.getValueCase() == ValueCase.LITERAL){
+              if(childExpression.getLiteral().getValue().getString().split("[:]")[1].equals("SECONDS")){
+                period = childExpression.getLiteral().getValue().getString().split("[:]")[0];
+              }
+            }
+          }
+
+          if(column != null && period != null){
+            timeSeriesColumnMap.put(column,period);
+          }
+        }
+      }
+    }
+  }
+
+  private void setTimeFilterMap(QueryRequest request) {
+
+    if(request.getFilter().getChildFilterCount() > 0){
+      for(Filter filter : request.getFilter().getChildFilterList()){
```

Comment:
This should be doing a tree walk, filters are a tree. Both of the following would be valid ways of passing down time range:
```
time > x AND time < y
(time > x AND time < y) AND other = z
```
while this would not be
```
(time > x AND time < y) OR other = z
```

Reply:
Continued discussion in another thread.